File: Symfony/src/Entity/Paiement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\PaiementRepository;
use Doctrine\DBAL\Types\Types;

#[ORM\Entity(repositoryClass: "App\Repository\PaiementRepository")]
class Paiement
{
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "IDENTITY")]
    #[ORM\Column(name: "id", type: "integer", nullable: false)]
    private $id;

    #[ORM\Column(name: "montant", type: "float", precision: 10, scale: 0, nullable: false)]
    private $montant;

    #[ORM\Column(name: "heure_P", type: "datetime", nullable: false)]
    private $heureP;

    #[ORM\ManyToOne(targetEntity: "Reservation")]
    #[ORM\JoinColumn(name: "id_res", referencedColumnName: "id")]
    private $idRes;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getMontant(): ?float
    {
        return $this->montant;
    }

    public function setMontant(float $montant): static
    {
        $this->montant = $montant;

        return $this;
    }

    public function getHeureP(): ?\DateTimeInterface
    {
        return $this->heureP;
    }

    public function setHeureP(\DateTimeInterface $heureP): static
    {
        $this->heureP = $heureP;

        return $this;
    }

    public function getIdRes(): ?Reservation
    {
        return $this->idRes;
    }

    public function setIdRes(?Reservation $idRes): static
    {
        $this->idRes = $idRes;

        return $this;
    }
    public function __construct()
    {
        $this->heureP = new \DateTime(); // Set the default value to the current date and time
    }

}

File: JAVA/src/main/java/models/Paiement.java
package models;

import java.sql.Date;


public class Paiement {
    private int id;
    private int id_res;
    private float montant_t;


    private Date heure_P;



    public Paiement() {
    }

    public Paiement(float montant_t,  Date heure_P) {
        this.montant_t = montant_t;

        this.heure_P = heure_P;
    }

    public Paiement(int id, int id_res, float montant_t) {
        this.id = id;
        this.id_res = id_res;
        this.montant_t = montant_t;


    }

    public Paiement(float montant_t) {
        this.montant_t = montant_t;


    }

    public Paiement(int id, int id_res, float montantT, Date heureP) {
        this.id=id;
        this.id_res=id_res;
        this.montant_t=montantT;
        this.heure_P=heureP;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

   public int getId_res() {
        return id_res;
    }

    public void setId_res(int id_res) {
        this.id_res = id_res;
    }

    public float getMontant_t() {
        return montant_t;
    }

    public void setMontant_t(float montant_t) {
        this.montant_t = montant_t;
    }



    public Date getHeure_P() {
        return heure_P;
    }

    public void setHeure_P(Date heure_P) {
        this.heure_P = heure_P;
    }
}
